src/lib/userStorage/UserStorageClass.js

```diff
@@ -835,6 +845,8 @@ export class UserStorage {
         receiptHash: get(event, 'data.receiptEvent.txHash'),
         endpoint: {
           address: sender,
+          toAddress,
```

Comment:
I don't understand why all those changes are needed if you just need to provider wallet account address to the chat widget (which could be received form wallet, see above)

Reply:
We did the best we could to get the other address in a given transaction (deeper explanation above) with the time we had, we are happy to clean it up more.  But `ownerAddress` refers to the other party of a single transaction in a given GD wallet.  

Its was originally name to refer to the transaction owner, not the wallet owner.  Apologize if that is a bit confusing in this context.

Reply:
@cryptoKevinL the data.endpoint.address should always refer to the counterparty in send/receive events
you could just use it without all those additional fields been stored

Reply:
gotcha, if this is the case we should be able to use this instead.